js/tidex2.js

```diff
@@ -0,0 +1,960 @@
+                        'max': undefined,
+                    },
+                    'cost': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                },
+                'info': market,
+            });
+        }
+        return result;
+    }
+
+    parseTicker (ticker, market = undefined) {
+        //
+        // fetchTicker
+        //     {
+        //         "name":"YFI_USDT",
+        //         "bid":"21607.842223",
+        //         "ask":"21635.50514299",
+        //         "open":"21530.31",
+        //         "high":"22619.38",
+        //         "low":"21193.59",
+        //         "last":"21644.23",
+        //         "volume":"188.4555124",
+        //         "deal":"4075612.142786252",
+        //         "change":"1"
+        //     }
+        //
+        // fetchTickers
+        //     {
+        //         "at":1646289676,
+        //         "ticker":{
+        //             "name":"yfi_usdt",
+        //             "bid":"21937.47155034",
+        //             "ask":"21966.13380147",
+        //             "open":"21449.08",
+        //             "high":"22619.38",
+        //             "low":"21193.59",
+        //             "last":"21968.03",
+        //             "vol":"188.4452385",
+        //             "deal":"4075398.065557751",
+        //             "change":"2"
+        //         }
+        //     }
+        //
+        const timestamp = this.safeTimestamp (ticker, 'at');
+        if (timestamp !== undefined) {
+            ticker = this.safeValue (ticker, 'ticker');
+        }
+        const marketId = this.safeStringUpper (ticker, 'name');
+        const symbol = this.safeSymbol (marketId, market);
+        const last = this.safeString (ticker, 'last');
+        return this.safeTicker ({
+            'symbol': symbol,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'high': this.safeString (ticker, 'high'),
+            'low': this.safeString (ticker, 'low'),
+            'bid': this.safeString (ticker, 'buy'),
+            'bidVolume': undefined,
+            'ask': this.safeString (ticker, 'sell'),
+            'askVolume': undefined,
+            'vwap': undefined,
+            'open': undefined,
+            'close': last,
+            'last': last,
+            'previousClose': undefined,
+            'change': undefined,
+            'percentage': undefined,
+            'average': this.safeString (ticker, 'avg'),
+            'baseVolume': this.safeString (ticker, 'vol_cur'),
+            'quoteVolume': this.safeString (ticker, 'vol'),
+            'info': ticker,
+        }, market, false);
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        await this.loadMarkets ();
+        const response = await this.publicGetTickers (params);
+        const result = this.safeValue (response, 'result');
```

Comment:
we need to add a fallback object-type as the third argument when using `this.safeValue (...)` so line 370 needs to be (also style-wise I think its nicer to have the parsing *after* the commented example) so I would move that line to be just above the `return` statement in line 395
```JavaScript
const result = this.safeValue (response, 'result', {})
```